src/pudl/transform/eia861.py

```diff
@@ -523,7 +523,76 @@ def _ba_code_backfill(df):
     return ba_eia861_filled
 
 
+def _tidy_customer_class_dfs(df, idx_cols):
+    # Clean up values just enough to use primary key columns as a multi-index:
+    # Transform values with fix_eia_na()
+    logger.debug("Cleaning table index columns so we can tidy data.")
+    raw_df = (
+        df.assign(
+            balancing_authority_code_eia=lambda x: x.balancing_authority_code_eia.fillna("UNK"))
+        .dropna(subset=["utility_id_eia"])
+        .astype({"utility_id_eia": pd.Int64Dtype()})
+        .set_index(idx_cols)
+    )
+    # Split the table into index, data, and "denormalized" columns for processing:
+    # Separate customer classes and reported data into a hierarchical index
+    logger.debug("Stacking EIA861 data columns by customer class.")
+    data_cols = _filter_customer_cols(raw_df, CUSTOMER_CLASSES)
+    data_cols.columns = (
+        data_cols.columns.str.split("_", n=1, expand=True)
+        .set_names(["customer_class", None])
+    )
+    # Now stack the customer classes into their own categorical column,
+    data_cols = (
+        data_cols.stack(level=0, dropna=False)
+        .reset_index()
+    )
+    denorm_cols = _filter_non_customer_cols(
+        raw_df, CUSTOMER_CLASSES).reset_index()
+
+    # Merge the index, data, and denormalized columns back together
+    tidy_df = pd.merge(denorm_cols, data_cols, on=idx_cols)
+
+    # Remove the now redundant "Total" records -- they can be reconstructed
+    # from the other customer classes.
+    tidy_df = _remove_total_col(tidy_df)
+
+    return tidy_df
+
+
+def _remove_total_col(df):
+    """Remove totals from customer class."""
+    df = df.query("customer_class!='total'")
+    return df
```

Comment:
Probably best to avoid functions that have only a single statement, unless it's a super opaque thing that becomes more readable by virtue of having a readable function name -- in this case just having a `query()` or `drop()` call inline is probably better.

Reply:
Good call!